src/ruvsarpur.py

```diff
@@ -439,8 +452,8 @@ def runMain():
         # Store the id as already recorded 
         previously_recorded.append(item['pid'])
 
-    # Now save the list of already recorded shows back to file and exit
-    savePreviouslyRecordedShows(previously_recorded)
+        # Now save the list of already recorded shows back to file and exit
```

Comment:
I did this in order to preserve the work already done.  That is, let's say you are downloading 10 episodes and, for some reason, the script aborts before you finish all ten.  This means that when you execute the download again, then the script will try to download the episodes that were fully downloaded prior to the previous temination thereby increasing download times and bandwidth usage.